pkg/syncer/syncer.go

```diff
@@ -103,6 +120,28 @@ func (c *Controller) process(gvr schema.GroupVersionResource, obj interface{}) e
 	return err
 }
 
+func (c *Controller) ensureNamespaceExists(namespace string) error {
+	klog.Infof("Ensuring namespace %s exists", namespace)
```

Comment:
This will probably end up being spammy, since we do this check before each new object is created or updated.

Can we perhaps only log when the namespace wasn't found?

Or, since if the namespace isn't found we'll find out anyway when we try to create-or-update the object, what's the value of this explicit namespace existence check?

Reply:
Since namespace existence is a prerequisite for request creation, creating it (if not already there) seemed natural to me.
The log here is obviously a rest from my tests.
I'll change remove it and will complete the error message in case the namespace cannot be created though it didn't already exist.